android/src/main/java/com/facebook/reactnative/androidsdk/FBProfileModule.java

```diff
@@ -18,24 +18,33 @@
     public static final String NAME = "FBProfile";
 
     public FBProfileModule(ReactApplicationContext reactContext) {
-      super(reactContext);
+        super(reactContext);
     }
 
     @NonNull
     @Override
     public String getName() {
-    return NAME;
-  }
+        return NAME;
+    }
 
     /**
-    * Get the current logged profile.
-    * @param callback Use callback to pass the current logged profile back to JS.
-    */
+     * Get the current logged profile.
+     *
+     * @param callback Use callback to pass the current logged profile back to JS.
+     */
     @ReactMethod
     public void getCurrentProfile(Callback callback) {
         //Return the profile object as a ReactMap.
-        callback.invoke(Profile.getCurrentProfile() == null
-                ? null
-                : Utility.profileToReactMap(Profile.getCurrentProfile()));
-  }
+        if (Profile.getCurrentProfile() == null) {
+            mProfileTracker = new ProfileTracker() {
+                @Override
+                protected void onCurrentProfileChanged(Profile oldProfile, Profile currentProfile) {
+                    callback.invoke(Utility.profileToReactMap(currentProfile));
+                    mProfileTracker.stopTracking();
+                }
+            };
```

Comment:
```suggestion
            AtomicBoolean hasInvoked = new AtomicBoolean(false);
            new ProfileTracker() {
                @Override
                protected void onCurrentProfileChanged(Profile oldProfile, Profile currentProfile) {
                    if (hasInvoked.compareAndSet(false, true)) {
                      callback.invoke(Utility.profileToReactMap(currentProfile));
                    }
                    mProfileTracker.stopTracking();
                }
            };
            try { Thread.currentThread().sleep(3000); } catch (InterruptedException e) { /* do nothing */ }
            hasInvoked.compareAndSet(false, true) {
              callback.invoke(null);
            }
```


Or similar, but I haven't compiled this either, as mentioned in the related issue it's not a use case of mine so I don't have time to pull it and test it, but something like this is needed I think, otherwise I think there will be issues in the future related to promises that don't resolve sometimes...

What do you think?